### ISSUE TITLE
ETH/AMPL Uniswap v2 LP problems

### ISSUE THREAD

**Issue by tr007008** [yam-finance/yam-protocol#66]
Hello,  
I'm having a bit of a problem staking Uniswap v2 tokens.  I seem to be staked but am hardly receiving any rewards.  

I actually staked two piles of tokens at ~around~ 8pm UTC yesterday.  The first pile was about 650 LINK tokens which right now have racked about 30 tokens.  

On the other hand, I also staked some ETH/AMPL Uniswap v2 LP tokens which at the time of staking, had a dollar value in the $8,000 range.  At this point, it has amassed a whopping 0.000001744 YAM.  

At first I thought it was a rebasing issue.  The LINK I managed to stake, I believe, right before the rebase.  The ETH/AMPL I did a few minutes later and likely landed sometime after the rebase.  So I waited to see if things changed over time.  It's been over 12 hours and still the same low reward.  That is to say, the number has technically increased, but by the same microscopic rate.

Anyone have any ideas what could have gone wrong?

**Comment by tr007008** [comment]
Okay, just for the sake of anyone else that might be guilty of the same lack of perception, I'm updating my own post.  It took a day or so of head-scratching for me to figure out what the deal was.  If it helps anyone else, then good.

The whole reason, of course, to have the Uniswap/ETH/AMPL token is to participate in the geyser.  I was, and still am a participant there.  What I didn't realize was that when I deposited to the geyser for some reason it didn't take the whole token (when you add to Uniswap's Liquidity Pool they issue you a token...actually it's a small fraction of a token, I guess it represents your percentage of the overall LP).   It took /most/ of it and left some "dust" in my wallet.

Long story short, when I went to stake in the YAM proj., it was fine letting me deposit a micro-amount.  It really would be nice if in V2/V3 there was some kind of warning to the effect of "The amount you're about to stake is about a hundredth of what you'll likely pay in gas...ARE YOU SURE??"

Another $25 error.  Guess I'll add it to the pile of other slippage/time-out/UniswapV2Router and other misc. errors that bleed away our precious ETH.  Sure will be nice when there's layer2 and sharding and etc and we can look back on all this and (maybe) laugh about the bad old days.

Good luck everyone -- be careful!

**Comment by Obmorok070** [comment]
> Hello,  
> 
> I'm having a bit of a problem staking Uniswap v2 tokens.  I seem to be staked but am hardly receiving any rewards.  
> 
> 
> 
> I actually staked two piles of tokens at around 8pm UTC yesterday.  The first pile was about 650 LINK tokens which right now have racked about 30 tokens.  
> 
> 
> 
> On the other hand, I also staked some ETH/AMPL Uniswap v2 LP tokens which at the time of staking, had a dollar value in the $8,000 range.  At this point, it has amassed a whopping 0.000001744 YAM.  
> 
> 
> 
> At first I thought it was a rebasing issue.  The LINK I managed to stake, I believe, right before the rebase.  The ETH/AMPL I did a few minutes later and likely landed sometime after the rebase.  So I waited to see if things changed over time.  It's been over 12 hours and still the same low reward.  That is to say, the number has technically increased, but by the same microscopic rate.
> 
> 
> 
> Anyone have any ideas what could have gone wrong?  
> 
> 
> 
>
